dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/css/OpenAPIUIBundlesListener.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.openapi20.internal.css;
+
+import java.util.Set;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceListener;
+import org.osgi.framework.ServiceReference;
+
+/**
+ * ServiceListener implementation that alerts OpenAPIUIBundleUpdater when all
+ * the expected bundles are ready
+ */
+public class OpenAPIUIBundlesListener implements ServiceListener {
+
+    private final Set<String> expectedBundleNames;
+
+    public OpenAPIUIBundlesListener(Set<String> expectedBundleNames) {
+        this.expectedBundleNames = expectedBundleNames;
+    }
+
+    @Override
+    public void serviceChanged(ServiceEvent event) {
```

Comment:
`serviceChanged` needs to be thread safe.  Right now I expect it is not with respect to manipulating the `expectedBundleNames` set.  Once you move to using a `CountDownLatch` you no longer are going to need to modify the set to remove the bundle names.  It will just be:

```java
            if (expectedBundleNames.contains(bundleName))
                expectedBundleNameLatch.countDown();
```
And you will not need any `unlockThread` method on the updater.

Reply:
Actually you probably do need to still remove from a set otherwise you could risk doing a `countDown` multiple times for the same bundleName.